### PR TITLE
bot: only dismiss reviewers if we have 2+ internal approvals

### PR DESCRIPTION
**Pull request by zmb3**

### CODE REVIEW

File: .github/workflows/robot/internal/bot/check_test.go
Comment (espadolini) [inline .github/workflows/robot/internal/bot/check_test.go:121]:
```suggestion
```

File: .github/workflows/robot/internal/bot/check.go
Comment (espadolini) [inline .github/workflows/robot/internal/bot/check.go:140]: nit: replace the string literal with the `approved` const in `robot/internal/review`?

If we export those, same goes for the test cases in `check_test` I guess.

**Review by espadolini — APPROVED** [review]

File: .github/workflows/robot/internal/bot/check_test.go
Comment (zmb3) [reply .github/workflows/robot/internal/bot/check_test.go:121]: Unfortunately required - `CheckAndSetDefaults` will fail otherwise.

File: .github/workflows/robot/internal/bot/check.go
Comment (zmb3) [reply .github/workflows/robot/internal/bot/check.go:140]: Will do.

Comment (nklaassen) [inline .github/workflows/robot/internal/bot/check.go:140]: if the same author approves twice does it get double counted?

Comment (espadolini) [reply .github/workflows/robot/internal/bot/check.go:140]: Reviews are sorted by time in github.go but they're not deduplicated by author, that's a good point.

We should reuse `review.reviewsByAuthor` here.

Comment (zmb3) [reply .github/workflows/robot/internal/bot/check.go:140]: Ughhh, great catch.

This seems unlikely, but worth fixing.

Comment (espadolini) [reply .github/workflows/robot/internal/bot/check.go:140]: It would only be an issue if an admin approver approved twice

Comment (zmb3) [reply .github/workflows/robot/internal/bot/check.go:148]: Should be fixed either way now.

**Review by espadolini — APPROVED** [review]

File: .github/workflows/robot/internal/bot/check.go
Comment (nklaassen) [reply .github/workflows/robot/internal/bot/check.go]: and what happens now if an approver comments after approving (such as https://github.com/gravitational/teleport/pull/14434#pullrequestreview-1037897066)? We have a helper `reviewsByAuthor` which handles that case, could be useful?

Comment (nklaassen) [reply .github/workflows/robot/internal/bot/check.go]: worst case we won't dismiss reviewers when we could have in this case, approving b/c that's better than what we have now

**Review by nklaassen — APPROVED** [review]

File: .github/workflows/robot/internal/bot/check.go
Comment (zmb3) [reply .github/workflows/robot/internal/bot/check.go]: Decided to leave this as is for now to fix the issue at hand.

Worst case, if someone comments after approval, we won't count it, and we may not dismiss reviewers when we could.

This is relatively uncommon, and if it does happen, we'll be no worse off than we are now. Having too many reviewers is better than having not enough.

**Review by russjones — APPROVED** [review]

**Pull request merged**
